graphql_compiler/interpreter/expression_ops/dispatch.py

```diff
@@ -30,7 +31,8 @@ def evaluate_expression(
     adapter: InterpreterAdapter[DataToken],
     query_metadata_table: QueryMetadataTable,
     query_arguments: Dict[str, Any],
-    current_type_name: str,
+    current_location: Optional[Location],
+    current_type_name: str,  # TODO remove this?
```

Comment:
This should be safe to remove now. The only place where the current type name would not match the location's type is during `can_coerce_to_type` where the current type name is the pre-coercion type (the `coerced_from_type` attribute instead of the `type` attribute on `LocationInfo`), but no expressions can exist during `can_coerce_to_type` and the type coercion happens before any filtering / outputting.

Feel free to change this in the next PR if you'd prefer not to grow this one any more.

Reply:
Sure. Wrote a clearer todo